src/coreclr/vm/precode_portable.hpp

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#ifndef __PRECODE_PORTABLE_H__
+#define __PRECODE_PORTABLE_H__
+
+#ifndef FEATURE_PORTABLE_ENTRYPOINTS
+#error Requires FEATURE_PORTABLE_ENTRYPOINTS to be set
+#endif // !FEATURE_PORTABLE_ENTRYPOINTS
+
+class PortableEntryPoint final
+{
+public: // static
+    static bool IsNativeEntryPoint(TADDR addr);
+    static void* GetActualCode(TADDR addr);
+    static MethodDesc* GetMethodDesc(TADDR addr);
+    static void* GetInterpreterData(TADDR addr);
+    static void SetInterpreterData(TADDR addr, PCODE interpreterData);
+
+private: // static
+    static PortableEntryPoint* ToPortableEntryPoint(TADDR addr);
+
+private:
+    void* _pActualCode;
+    MethodDesc* _pMD;
+    void* _pInterpreterData;
+
+    // We keep the canary value last to ensure a stable ABI across build flavors
+    INDEBUG(size_t _canary);
+
+public:
+    void Init(MethodDesc* pMD);
+};
+
+extern InterleavedLoaderHeapConfig s_stubPrecodeHeapConfig;
+
+enum PrecodeType
+{
+    PRECODE_INVALID = -100,
+    PRECODE_STUB,
+    PRECODE_UMENTRY_THUNK,
+    PRECODE_FIXUP,
+    PRECODE_PINVOKE_IMPORT,
+    PRECODE_INTERPRETER,
```

Comment:
Do we want to make this FEATURE_INTERPRETER conditional?

Reply:
I don't think so. These are all placeholders that should eventually be removed. They are simply here to make compiling possible. When I remove codepaths I remove the nonsense here.